src/Model/Resolver/MoveWishlistToCart.php

```diff
@@ -95,29 +96,56 @@ public function __construct(
      */
     protected function addItemsToCart(array $wishlistItems, CartInterface $quote): void
     {
-        foreach ($wishlistItems as $item) {
-            $product = $item['product'];
+        $cartItems = $quote->getItems();
+        $addedQty = false;
+        $shouldShowError = false;
+
+        foreach ($cartItems as $item) {
+            $product = $item->getProduct();
+            $qty = $item->getQty();
+            $sku = $product->getSku();
+
+            if (array_key_exists($sku, $wishlistItems)) {
+                $wishlistItem = $wishlistItems[$sku];
+                unset($wishlistItems[$sku]);
 
-            $stockStatus = $this->stockStatusRepository->get($product->getId());
-            $productStockStatus = $stockStatus->getStockStatus();
+                $qty += $wishlistItem['qty'];
+                $item->setQty($qty);
 
-            // Don't add out of stock product to cart
-            if($productStockStatus === StockStatusInterface::STATUS_OUT_OF_STOCK){
-                continue;
+                $wishlistItem['item']->delete();
+                $addedQty = true;
             }
+        }
 
-            $data = json_decode($item['buy_request'], true);
+        $allItemsStatus = false;
```

Comment:
This naming is not verbose - it is impossible to understand what this is a flag for. Please rename.